lib/task.js

```diff
@@ -268,29 +280,37 @@ export async function getUnfinishedTasks() {
  * @returns {NamedNode} The URI of the vendor.
  */
 export async function getVendor(task) {
-  const response = await mas.querySudo(`
-    ${cts.SPARQL_PREFIXES}
+  const response = await querySudo(`
+    ${SPARQL_PREFIXES}
     SELECT ?vendor {
-      ${rst.termToString(task)} dct:isPartOf ?job .
+      ${termToString(task)} dct:isPartOf ?job .
       ?job
         rdf:type cogs:Job ;
         prov:wasAssociatedWith ?vendor .
     }
     LIMIT 1
   `);
-  const parser = new sjp.SparqlJsonParser();
+  const parser = new SparqlJsonParser();
   return parser.parseJsonResults(response)[0]?.vendor;
 }
 
 export async function waitForDatabase() {
-  // eslint-disable-next-line no-constant-condition
-  while (true) {
+  const maxRetries = 30;
```

Comment:
instead of indefinite waiting for the DB, changed it to about 1 minute and raising an error after that